src/raster/rasterband.rs

```diff
@@ -863,6 +954,50 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: i32,
+    histogram: *mut i32,
+    histogram_vec: Option<Vec<i32>>,
```

Comment:
I personally think the counts should be `usize` or `u32`, regardless of what GDAL C returns, but not sure that's a consensus opinion. @lnicola @jdroenner you opinion?